src/server/frontend_xwayland/xwayland_default_configuration.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2018 Marius Gripsgard <marius@ubports.com>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "mir/default_server_configuration.h"
+#include "mir/log.h"
+#include "wayland_connector.h"
+#include "xwayland_connector.h"
+
+#include <string>
+
+#include "mir/options/default_configuration.h"
+
+namespace mf = mir::frontend;
+
+std::shared_ptr<mf::Connector> mir::DefaultServerConfiguration::the_xwayland_connector()
+{
+    return xwayland_connector([this]() -> std::shared_ptr<mf::Connector> {
+        // For now just use a env var to set X11 display
+        // If not set use 0
+        int disp = 0;
+        try
+        {
+            disp = std::stoi(getenv("MIR_X11_DISPLAY"));
+        }
+        catch (...)
+        {
+            mir::log_error("No valid x11 display vaule provided, must be number only!");
```

Comment:
Not something that needs fixing to land this code, but:

1. This ought to be a configuration option; and,
2. If not provided (the default) then we should not provide X11 support.

Reply:
1. yeah, makes sense. but keeping the envar would be nice too (for scripts etc)
2. I would more say, if not provided, find next available x11 disp.

Reply:
1. Mir config comes from any of three places: command-line, env variables or config file (in that order). So making a config option supports all three. E.g. we can say ":3" three ways:
```
$ miral-shell --x11-desktop :3
$ MIR_SERVER_X11_DISPLAY=:3 miral-shell
$ echo x11-desktop=:3 > ~/.config/miral-shell.config && miral-shell
```

2. No. If I don't ask for X11 I don't want X11. There's too much second-guessing the user already,